src/main/java/com/strandls/dataTable/service/impl/DataTableServiceImpl.java

```diff
@@ -242,4 +301,38 @@ private DataTable dataTableSerilizer(DataTableWkt dt, DataTable prevState) {
 
 	}
 
+	@Override
+	public Activity addDataTableComment(HttpServletRequest request, CommentLoggingData comment) {
+		try {
+			comment.setMailData(generateMailData(request,comment.getRootHolderId()));
+			activityService = headers.addActivityHeaders(activityService, request.getHeader(HttpHeaders.AUTHORIZATION));
+			Activity result = activityService.addComment("datatable", comment);
+			return result;
+		} catch (Exception e) {
+			logger.error(e.getMessage());
+		}
+		return null;
+	}
+
+	@Override
+	public List<UserGroupIbp> updateUserGroupDatatableMapping(HttpServletRequest request, Long datatableId,
+			UserGroupCreateDatatable userGroups) {
+		DataTable dataTable = new DataTable();
+		try {
+			dataTable = dataTableDao.findById(datatableId);
```

Comment:
same as above pls add null pointer checks after find by id, i case a wrng id is sent to the backend it would throw a null pointer error